### DISCUSSION TITLE
Responsive grid with form elements

### DISCUSSION THREAD

**Discussion by r100-stack** [iTwin/iTwinUI#2086] [Examples]
@FlyersPh9, @Ben-Pusey-Bentley, and I noticed that creating a responsive grid of form elements was a common UI pattern. So, we created a small sample of *one* of the ways to implement this:

[Sample sandbox](https://stackblitz.com/edit/github-m7zngb-8lmck4)

Responsive design: Uses arbitrary breakpoints of 600px and 400px:

|width > 600px|600px > width > 400px|400px > width|
|-|-|-|
|<img width="715" alt="image" src="https://github.com/iTwin/iTwinUI/assets/45748283/f9e879af-aef6-489e-bc33-303102f0cf13">|<img width="477" alt="image" src="https://github.com/iTwin/iTwinUI/assets/45748283/0d423c14-b1f5-4670-a7a8-39fb018f68c1">|<img width="350" alt="image" src="https://github.com/iTwin/iTwinUI/assets/45748283/03cb149f-e4e8-46fd-89ff-77deb479474d">|

<details>
<summary>Screen recording</summary>

https://github.com/iTwin/iTwinUI/assets/45748283/facd28fb-e519-4af8-abe2-762aa061882e

</details>

**Comment by mayank99** [comment]
This should probably use container queries instead of media queries. The screen-size is irrelevant for forms that may be nested deep inside a small dialog or widget or something.
